templates/fine-tune-function-calling/README.md

```diff
@@ -0,0 +1,419 @@
+
+
+```python
+# to be run only if you finetuned on the Anyscale platform
+ANYSCALE_API_KEY="service-api-key-here"
+ANYSCALE_API_BASE="service-url-here" 
+if not ANYSCALE_API_BASE.endswith("/"):
+    ANYSCALE_API_BASE += "/"
+ANYSCALE_API_BASE += "v1"
+# enter the model id here. This would be different depending on whether you performed LoRA or full parameter fine-tuning.
+# Example: meta-llama/Meta-Llama-3-8B-Instruct:mysuffix:myid 
+MODEL_ID = "ModelIdHere"
+```
+
+## Step 3(b): Finetuned through serverless endpoints
+
+To serve the fine-tuned model, you just need to navigate to the "Serving" section on the Anyscale Platform. Your fine-tuned model should already be visible in the list of available models! 
+
+<p align="center">
+  <img src="./assets/serving_endpoints.png" alt="Serve Endpoints">
+</p>
+
+
+As in the above image, click on the three dots and then click on "Query". This will provide you the starter code to interact with the model via curl, python, etc. 
+
+<p align="center">
+  <img src="./assets/serve_api_key.png" alt="API Key">
+</p>
+
+
+
+```python
+## This is only if you finetuned through serverless endpoints
+ANYSCALE_API_BASE = "https://api.endpoints.anyscale.com/v1"
+ANYSCALE_API_KEY = "esecret_yourKeyHere"
+MODEL_ID = "yourModelIdHere" # make sure to not add a stray slash "/"" at the end!
+```
+
+### (Optional) Try out the model via Playground
+
+You can try out your new model in the Playground: https://console.anyscale.com/v2/playground . In the model dropdown, you should be able to see your finetuned model as shown below
+
+<p align="center">
+  <img src="./assets/playground.png" alt="Playground">
+</p>
+
+# Step 4: Evaluation
+
+Let's evaluate our trained model with GPT-4 as a baseline. 
+
+
+## Evaluation strategy
+
+Evaluation of function calling capability is non-trivial, given that we're looking to extract structured data from an inherently unpredictable and unstructured stream of text. We will use the following simple evaluation strategy: The models are evaluated on the accuracy metric and their responses are graded as accurate if their response for each assistant entry in the conversation is correct. An assistant response is graded as correct under the below conditions:
+1. In case the ground truth response contains no function call, then the model's response should also not have a function call. 
+2. In case the ground truth response contains a function call, then the model's response should also have a function call. We do not check for the content of the response here. The assistant function call should further have the correct function name and the correct function arguments. 
```

Comment:
> We do not check for the content of the response here.
This is confusing? I know you mean the "content" in the message but instead of this just say we only check the name and arguments of the function call to ensure correctness.

Reply:
Have fixed the wording now.